test/OrchardCore.Tests/Apis/Context/SiteStartup.cs

```diff
@@ -1,60 +1,34 @@
 using OrchardCore.Modules;
-using OrchardCore.Modules.Manifest;
 using OrchardCore.Recipes.Services;
+using OrchardCore.Testing;
+using OrchardCore.Testing.Apis;
+using OrchardCore.Testing.Apis.Security;
+using OrchardCore.Testing.Recipes;
+using OrchardCore.Testing.Stubs;
 
 namespace OrchardCore.Tests.Apis.Context
 {
     public class SiteStartup
     {
-        public static ConcurrentDictionary<string, PermissionsContext> PermissionsContexts;
-
-        static SiteStartup()
-        {
-            PermissionsContexts = new ConcurrentDictionary<string, PermissionsContext>();
-        }
-
         public void ConfigureServices(IServiceCollection services)
         {
-            services.AddOrchardCms(builder =>
-                builder.AddSetupFeatures(
-                    "OrchardCore.Tenants"
-                )
-                .AddTenantFeatures(
-                    "OrchardCore.Apis.GraphQL"
-                )
-                .ConfigureServices(collection =>
+            services.AddOrchardCms(builder => builder
+                .AddSetupFeatures("OrchardCore.Tenants")
+                .AddTenantFeatures("OrchardCore.Apis.GraphQL")
+                .ConfigureServices(serviceCollection =>
                 {
-                    collection.AddScoped<IRecipeHarvester, TestRecipeHarvester>();
+                    serviceCollection.AddScoped<IRecipeFileProvider, RecipeFileProvider>();
+                    serviceCollection.AddScoped<IRecipeHarvester, RecipeHarvesterStub>();
 
-                    collection.AddScoped<IAuthorizationHandler, PermissionContextAuthorizationHandler>(sp =>
-                    {
-                        return new PermissionContextAuthorizationHandler(sp.GetRequiredService<IHttpContextAccessor>(), PermissionsContexts);
-                    });
+                    serviceCollection.AddScoped<IAuthorizationHandler, PermissionContextAuthorizationHandler>(sp =>
+                        new PermissionContextAuthorizationHandler(sp.GetRequiredService<IHttpContextAccessor>(),
+                        SiteContextOptions.PermissionsContexts));
                 })
                 .Configure(appBuilder => appBuilder.UseAuthorization()));
 
             services.AddSingleton<IModuleNamesProvider, ModuleNamesProvider>();
```

Comment:
services.AddSingleton<IModuleNamesProvider>(new ModuleNamesProvider(typeof(Program).Assembly));

pass the Assembly to ModuleNamesProvider